pkg/actuators/client/client.go

```diff
@@ -20,86 +20,407 @@ import (
 	if err != nil {
 		return nil, err
 	}
 
+	// Get Resource Group ID
+	resourceGroupName := machineProviderConfig.ResourceGroup
+	resourceGroupID, err := c.GetResourceGroupIDByName(resourceGroupName)
+	if err != nil {
+		return nil, err
+	}
+
+	// Get Custom Image ID
+	imageID, err := c.GetCustomImageByName(machineProviderConfig.Image, resourceGroupID)
+	if err != nil {
+		return nil, err
+	}
+
+	// Get VPC ID
+	vpcName := machineProviderConfig.VPC
+	vpcID, err := c.GetVPCIDByName(vpcName, resourceGroupID)
+	if err != nil {
+		return nil, err
+	}
+
+	// Get Subnet ID
+	subnetName := machineProviderConfig.PrimaryNetworkInterface.Subnet
+	subnetID, err := c.GetSubnetIDbyName(subnetName, resourceGroupID)
+	if err != nil {
+		return nil, err
+	}
+
+	// Get Security Groups
+	securityGroups, err := c.GetSecurityGroupsByName(machineProviderConfig.PrimaryNetworkInterface.SecurityGroups, resourceGroupID, vpcID)
+	if err != nil {
+		return nil, err
+	}
+
+	// Create Instance Options
+	options := &vpcv1.CreateInstanceOptions{}
+
+	// Set Instance Prototype - Contains all the info necessary to provision an instance
+	options.SetInstancePrototype(&vpcv1.InstancePrototype{
+		Name: &machineName,
+		Image: &vpcv1.ImageIdentity{
+			ID: &imageID,
+		},
+		Profile: &vpcv1.InstanceProfileIdentity{
+			Name: &machineProviderConfig.Profile,
+		},
+		Zone: &vpcv1.ZoneIdentity{
+			Name: &machineProviderConfig.Zone,
+		},
+		ResourceGroup: &vpcv1.ResourceGroupIdentity{
+			ID: &resourceGroupID,
+		},
+		PrimaryNetworkInterface: &vpcv1.NetworkInterfacePrototype{
+			Subnet: &vpcv1.SubnetIdentity{
+				ID: &subnetID,
+			},
+			SecurityGroups: securityGroups,
+		},
+		VPC: &vpcv1.VPCIdentity{
+			ID: &vpcID,
+		},
+		UserData: &userData,
+	})
+
+	// Create a new Instance from an instance prototype object
+	instance, _, err := c.vpcService.CreateInstance(options)
+	if err != nil {
+		return nil, err
+	}
+
+	return instance, nil
+}
+
+// GetVPCIDByName Retrives VPC ID
+func (c *ibmCloudClient) GetVPCIDByName(vpcName string, resourceGroupID string) (string, error) {
+	// Initialize List Vpcs Options
+	vpcOptions := c.vpcService.NewListVpcsOptions()
+
+	// Set Resource Group ID
+	vpcOptions.SetResourceGroupID(resourceGroupID)
+
+	// Get a list all VPCs
+	vpcList, _, err := c.vpcService.ListVpcs(vpcOptions)
+	if err != nil {
+		return "", err
+	}
+
+	if vpcList != nil {
+		var vpcID string
+		for _, eachVPC := range vpcList.Vpcs {
+			if *eachVPC.Name == vpcName {
+				vpcID = *eachVPC.ID
+				return vpcID, nil
+			}
+		}
+	}
+
+	return "", fmt.Errorf("Could not retrieve VPC ID of name: %v", vpcName)
+}
+
+// GetCustomImageByName retrieves custom image from VPC by region and name
+func (c *ibmCloudClient) GetCustomImageByName(imageName string, resourceGroupID string) (string, error) {
+	// Initialize List Images Options
+	options := c.vpcService.NewListImagesOptions()
+
+	// Private images
+	options.SetVisibility(vpcv1.ImageVisibilityPrivateConst)
```

Comment:
Wondering if setting the visibility filter here is necessary?
The resource group and name filters should yield a single image result.
Should vpc ever change the visibility options, add support for non-private custom images, or provide a non-private rhcos image for use in the future, this setting could potentially require adjustment later on.
i.e. does it buy us anything?

Reply:
Added that just to keep it aligned with VPC custom image creation. Any custom image we create is `private`. Not quite sure what the default visibility is when we pull a `List of images` through SDK, but if it's set to `public`, we will not get the image. So setting it should be better so we know we are always pulling `private` image. 

If VPC ever allows making that change while creating a custom image, we will still be good unless the default becomes `public` during that creation and we are not setting the `visibility` - at that point will need to update Terraform scripts as well.